### PR TITLE
FR-1433: General Application Outcome

### PR DESCRIPTION
**Pull request by HarryH96**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FR-1433
Linked to Notification Service -> https://github.com/hmcts/finrem-notification-service/pull/174

### Change description ###
Triggering Email for General Application Outcome

NOTE - I have branched off from Shauns PR for https://tools.hmcts.net/jira/browse/FR-1805

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnoneill46 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 1]
```diff
@@ -189,6 +189,12 @@ public void sendContestedGeneralApplicationReferToJudgeEmail(CallbackRequest cal
         sendNotificationEmail(notificationRequest, uri);
     }
 
+    public void sendContestedGeneralApplicationOutcomeEmail(CallbackRequest callbackRequest) {
+        URI uri = buildUri(notificationServiceConfiguration.getContestedGeneralApplicationOutcome());
+        notificationRequest = createNotificationRequest(callbackRequest);
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:194]: do you not need to set the notification email to FRC here?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 2]
```diff
@@ -173,22 +183,49 @@ public void sendContestedConsentOrderNotApprovedEmail(CallbackRequest callbackRe
...
+
+        Map<String, Object> data = callbackRequest.getCaseDetails().getData();
+        Map<String, Object> courtDetailsMap = objectMapper.readValue(getCourtDetailsString(), HashMap.class);
+        Map<String, Object> courtDetails = (Map<String, Object>) courtDetailsMap.get(data.get(CaseHearingFunctions.getSelectedCourt(data)));
+
+        FrcCourtDetails selectedFRCDetails = FrcCourtDetails.builder()
+            .courtName((String) courtDetails.get(COURT_DETAILS_NAME_KEY))
+            .courtAddress((String) courtDetails.get(COURT_DETAILS_ADDRESS_KEY))
+            .phoneNumber((String) courtDetails.get(COURT_DETAILS_PHONE_KEY))
+            .email((String) courtDetails.get(COURT_DETAILS_EMAIL_KEY))
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:212]: I think you can get away with doing `(String) courtDetails.get(COURT_DETAILS_EMAIL_KEY)`, you don't use anything else

File: src/main/resources/json/court-details.json [hunk 3]
```diff
@@ -3,474 +3,474 @@
     "courtName": "Bromley County Court And Family Court",
     "courtAddress": "Bromley County Court, College Road, Bromley, BR1 3PX",
     "phoneNumber": "0208 290 9620",
-    "email": "family.bromley.countycourt@justice.gov.uk"
+    "email": "fr_applicant_sol@sharklasers.com"
```
Comment (shaunganley) [inline src/main/resources/json/court-details.json:6]: what's happening here? is this temporary?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java [hunk 4]
```diff
@@ -569,13 +599,22 @@ public void throwExceptionWhenGeneralEmailIsRequested() {
...
         Map<String, Object> caseData = new HashMap<>();
         caseData.put(CONTESTED_SOLICITOR_EMAIL, "test@test.com");
         caseData.put(CONTESTED_SOLICITOR_NAME, "solicitorName");
         caseData.put(SOLICITOR_REFERENCE, "56789");
         caseData.put(REGION, regionValue);
         caseData.put(frcList, frcValue);
+        //add the court list and the court value
+        caseData.put(courtList, courtValue);
+
+        /*
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java:613]: can you tidy this up

**Review by shaunganley — CHANGES_REQUESTED** [review]
few small comments

File: src/main/resources/json/court-details.json (hunk 3 above)
Comment (HarryH96) [reply src/main/resources/json/court-details.json:6]: Yeah temporary for Vivek/Me to test using this email, as we dont actually want to send this to an FRC just yet

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java (hunk 4 above)
Comment (HarryH96) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java:613]: Oops!

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java (hunk 2 above)
Comment (HarryH96) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:212]: Good spot

File: hs_err_pid67836.log [hunk 5]
```diff
@@ -0,0 +1,1167 @@
+#
```
Comment (shaunganley) [inline hs_err_pid67836.log:1]: WAT

File: src/main/resources/json/court-details.json (hunk 3 above)
Comment (shaunganley) [reply src/main/resources/json/court-details.json:6]: but we're not merging this right? did you add the toggle to send to this test address in nonprod envs?

File: hs_err_pid67836.log (hunk 5 above)
Comment (HarryH96) [reply hs_err_pid67836.log]: This file is created from the JRE Errors when running locally, didnt notice it! Ive removed the file and added .log files to the gitignore

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/FeatureToggleService.java [hunk 6]
```diff
@@ -66,6 +67,10 @@ public boolean isAutomateSendOrderEnabled() {
         return isFeatureEnabled(AUTOMATE_SEND_ORDER);
     }
 
+    public boolean isSendToFRCEnabled() {
+        return isFeatureEnabled(SEND_TO_FRC);
+    }
```
Comment (johnoneill46) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/FeatureToggleService.java:72]: Might just be worth adding a comment to this to explain exactly what it's for so people in future don't decide to remove it

File: src/main/resources/json/court-details.json (hunk 3 above)
Comment (HarryH96) [reply src/main/resources/json/court-details.json]: toggle has now been added!

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java [hunk 7]
```diff
@@ -958,6 +991,30 @@ public void throwExceptionWhenContestedGeneralApplicationReferToJudgeEmailIsRequ
...
     }
 
+    @Test
+    public void sendContestedGeneralApplicationOutcomeNotificationEmailWhenSendToFRCToggleTrue() throws IOException {
+        when(featureToggleService.isSendToFRCEnabled()).thenReturn(true);
+        callbackRequest = getContestedCallbackRequest(MIDLANDS, MIDLANDS_FRC_LIST,
+            NOTTINGHAM, NOTTINGHAM_COURTLIST,  "FR_s_NottinghamList_1");
+
+        mockServer.expect(MockRestRequestMatchers.requestTo(END_POINT_CONTESTED_GENERAL_APPLICATION_OUTCOME))
+            .andExpect(MockRestRequestMatchers.method(HttpMethod.POST))
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java:1001]: can you update these 2 days to expect the correct value for notification email as done here https://github.com/hmcts/finrem-case-orchestration-service/blob/426ee347632148725460b1957cb373a0a0dccf8b/src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationServiceTest.java#L969

**Review by shaunganley — APPROVED** [review]
1 small comment for tests

File: charts/finrem-cos/values.yaml [hunk 8]
```diff
@@ -8,7 +8,7 @@ java:
     IDAM_API_URL: "https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net"
     IDAM_S2S_URL : "http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
     DOCUMENT_GENERATOR_SERVICE_API_BASEURL : "http://finrem-dgcs-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
-    FINREM_NOTIFICATION_SERVICE_BASE_URL : "http://finrem-ns-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+    FINREM_NOTIFICATION_SERVICE_BASE_URL : "http://finrem-ns-pr-174.service.core-compute-preview.internal"
```
Comment (jakub-kozlowski) [inline charts/finrem-cos/values.yaml:11]: this needs to be backed out before merging?

**Review by jakub-kozlowski — CHANGES_REQUESTED** [review]

File: charts/finrem-cos/values.yaml (hunk 8 above)
Comment (HarryH96) [reply charts/finrem-cos/values.yaml]: Updating this now to merge 👍

**Pull request merged**